pkg/cloud/services/ec2/instances.go

```diff
@@ -1008,6 +1020,49 @@ func (s *Service) ModifyInstanceMetadataOptions(instanceID string, options *infr
 	return nil
 }
 
+// GetDHCPOptionSetDomainName returns the domain DNS name for the VPC from the DHCP Options.
+func (s *Service) GetDHCPOptionSetDomainName(ec2client ec2iface.EC2API, vpcID *string) *string {
+	fmt.Println("vpcID", vpcID)
+	log := s.scope.GetLogger()
+
+	if vpcID == nil {
+		log.Error(errors.New("vpcID is nil"), "vpcID is nil")
+		return nil
+	}
+
+	vpcInput := &ec2.DescribeVpcsInput{
+		VpcIds: []*string{vpcID},
+	}
+
+	vpcResult, err := ec2client.DescribeVpcs(vpcInput)
+	if err != nil {
+		return nil
+	}
+
+	dhcpInput := &ec2.DescribeDhcpOptionsInput{
+		DhcpOptionsIds: []*string{vpcResult.Vpcs[0].DhcpOptionsId},
+	}
+
+	log.Info("Fetching DHCP Options to get the domain-name", "dhcpOptionsID", *vpcResult.Vpcs[0].DhcpOptionsId)
+
+	dhcpResult, err := ec2client.DescribeDhcpOptions(dhcpInput)
+	if err != nil {
+		return nil
+	}
+
+	for _, dhcpConfig := range dhcpResult.DhcpOptions[0].DhcpConfigurations {
+		if *dhcpConfig.Key == "domain-name" {
+			for _, value := range dhcpConfig.Values {
+				if value.Value != nil && *value.Value != fmt.Sprintf("%s.compute.internal", s.scope.Region()) {
+					return aws.String(*value.Value)
+				}
+			}
+		}
+	}
```

Comment:
```suggestion
	for _, dhcpConfig := range dhcpResult.DhcpOptions[0].DhcpConfigurations {
		if *dhcpConfig.Key == "domain-name" {
			if len(dhcpConfig.Values) == 0 {
			   return nil
			}
			domainName := dhcpConfig.Values[0]
			// default domainName is 'ec2.internal' in us-east-1 and 'region.compute.internal' in the other regions.
			if strings.Contains(*domainName, "ec2.internal") || strings.Contains(*domainName, "compute.internal") {
			   return nil
			} 
			return domainName 
		}
	}
```
no need to loop over `dhcpConfig.Values`, it can only has 1 value.

Reply:
```
Key: "domain-name",
	          Values: [{
	              Value: "example.com"
	            }]
```
once you find the "domain-name key, the associated `Values` array field can only have 1 entry.

you still need to loop over `dhcpResult.DhcpOptions[0].DhcpConfigurations` thats right